src/request.rs

```diff
@@ -146,11 +118,12 @@ impl Request {
     /// }
     /// ```
     #[cfg(feature = "json")]
-    pub fn send_json(&mut self, data: SerdeValue) -> Result<Response> {
-        if self.header("Content-Type").is_none() {
-            self.set("Content-Type", "application/json");
+    pub fn send_json(self, data: SerdeValue) -> Result<Response> {
+        let mut this = self;
```

Comment:
You can avoid the `let mut this = self` by declaring self to be `mut` in the method signature:

```
pub fn send_json(self, data: SerdeValue) -> Result<Response> {
```

Reply:
Can you reassign self?

Reply:
I believe so! Haven't double-checked; I could be wrong.